File: license/java/license.java
public abstract class license {
    private final String name;
    private final String duration;
    private final double cost;

    public license(String name, String duration, double cost) {
        this.name = name;
        this.duration = duration;
        this.cost = cost;
    }

    public String getName() {
        return name;
    }

    public String getDuration() {
        return duration;
    }

    public double getCost() {
        return cost;
    }

    public abstract double CalculateCost();
}

File: license - copia/php/license.php
<?php

abstract class License {
    private string $name;
    private string $duration;
    private float $cost;

    public function __construct(string $name, string $duration, float $cost) {
        $this->name = $name;
        $this->duration = $duration;
        $this->cost = $cost;
    }

    public function getName(): string {
        return $this->name;
    }

    public function getDuration(): string {
        return $this->duration;
    }

    public function getCost(): float {
        return $this->cost;
    }

    public abstract function calculateCost(): float;
}

?>